transports/quic/src/connection.rs

```diff
@@ -183,16 +186,14 @@ impl Connection {
     /// On success, a [`quinn_proto::StreamEvent::Finished`] event will later be produced when the
     /// substream has been effectively closed. A [`ConnectionEvent::StreamStopped`] event can also
     /// be emitted.
-    pub fn shutdown_substream(
+    pub fn finish_substream(
```

Comment:
I think I prefer to stick with `finish_substream` because it is consistent if the name of the event it produces `quinn_proto::StreamEvent::Finished` / `ConnectionEvent::StreamFinished`. 
But not a strong opinion, happy to change it to `close_write` if you insist :)